make env errors out

```bash

"/usr/bin/python2.7" -c "import virtualenv" || pip install virtualenv
"/usr/bin/python2.7" -m virtualenv --no-download ./env/
Usage: virtualenv.py [OPTIONS] DEST_DIR

virtualenv.py: error: no such option: --no-download
make: *** [env] Error 2
```

> We can probably close

I actually get stuck on `make run` now.

```bash
PYTHONPATH=. env/bin/honcho run -e defaults.env,local.env env/bin/python liberapay/models/__init__.py
Traceback (most recent call last):
  File "liberapay/models/__init__.py", line 178, in <module>
    from liberapay import wireup
  File "liberapay.com/liberapay/wireup.py", line 21, in <module>
    import liberapay.billing.payday
  File "liberapay.com/liberapay/billing/payday.py", line 12, in <module>
    from liberapay.billing.exchanges import transfer
  File "liberapay.com/liberapay/billing/exchanges.py", line 27, in <module>
    from liberapay.models.participant import Participant
  File "liberapay.com/liberapay/models/participant.py", line 5, in <module>
    from hashlib import pbkdf2_hmac, md5
ImportError: cannot import name pbkdf2_hmac
make: *** [db-migrations] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -5,6 +5,7 @@ install_where := $(shell $(python) -c "import sys; print('' if hasattr(sys, 'rea
 # NOTE: Creating a virtualenv on Windows places binaries in the 'Scripts' directory.
 bin_dir := $(shell $(python) -c 'import sys; print("Scripts" if sys.platform == "win32" else "bin")')
 env_bin := env/$(bin_dir)
+env_py := $(env_bin)/python
 test_env_files := defaults.env,tests/test.env,tests/local.env
 pip := pip --disable-pip-version-check
 with_local_env := $(env_bin)/honcho run -e defaults.env,local.env
@@ -35,19 +36,19 @@ schema-diff: test-schema
 	rm prod.sql local.sql
 
 data: env
-	$(with_local_env) $(env_bin)/python -m liberapay.utils.fake_data
+	$(with_local_env) $(env_py) -m liberapay.utils.fake_data
 
 db-migrations: sql/migrations.sql
-	PYTHONPATH=. $(with_local_env) $(env_bin)/python liberapay/models/__init__.py
+	PYTHONPATH=. $(with_local_env) $(env_py) liberapay/models/__init__.py
 
 run: env db-migrations
-	PATH=$(env_bin):$$PATH $(with_local_env) $(env_bin)/python app.py
+	PATH=$(env_bin):$$PATH $(with_local_env) $(env_py) app.py
 
 py: env
-	PYTHONPATH=. $(with_local_env) $(env_bin)/python -i liberapay/main.py
+	PYTHONPATH=. $(with_local_env) $(env_py) -i liberapay/main.py
 
 payday: env
-	PYTHONPATH=. $(with_local_env) $(env_bin)/python liberapay/billing/payday.py
+	PYTHONPATH=. $(with_local_env) $(env_py) liberapay/billing/payday.py
 
 test-schema: env
 	$(with_tests_env) ./recreate-schema.sh test

--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,5 @@
 python := "$(shell { command -v python2.7 || command -v python; } 2>/dev/null)"
+install_where := $(shell $(python) -c "import sys; print('' if hasattr(sys, 'real_prefix') else '--user')")
 
 # Set the relative path to installed binaries under the project virtualenv.
 # NOTE: Creating a virtualenv on Windows places binaries in the 'Scripts' directory.
@@ -14,8 +15,8 @@ echo:
 	@echo $($(var))
 
 env: requirements*.txt
-	where=$$($(python) -c "import sys; print('' if hasattr(sys, 'real_prefix') else '--user')") \
-	$(python) -m $(pip) install $$where "virtualenv>=15.0.0"
+	$(python) -m ensurepip $(install_where)
+	$(python) -m $(pip) install $(install_where) "virtualenv>=15.0.0"
 	$(python) -m virtualenv --no-download ./env/
 	$(env_bin)/$(pip) install $$(for f in requirements*.txt; do echo "-r $$f"; done)
 	@touch env

--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@ python := "$(shell { command -v python2.7 || command -v python; } 2>/dev/null)"
 bin_dir := $(shell $(python) -c 'import sys; print("Scripts" if sys.platform == "win32" else "bin")')
 env_bin := env/$(bin_dir)
 test_env_files := defaults.env,tests/test.env,tests/local.env
-pip := $(env_bin)/pip --disable-pip-version-check
+pip := pip --disable-pip-version-check
 with_local_env := $(env_bin)/honcho run -e defaults.env,local.env
 with_tests_env := $(env_bin)/honcho run -e $(test_env_files)
 py_test := $(with_tests_env) $(env_bin)/py.test
@@ -14,9 +14,10 @@ echo:
 	@echo $($(var))
 
 env: requirements*.txt
-	$(python) -m pip --disable-pip-version-check install --user "virtualenv>=15.0.0"
+	where=$$($(python) -c "import sys; print('' if hasattr(sys, 'real_prefix') else '--user')") \
+	$(python) -m $(pip) install $$where "virtualenv>=15.0.0"
 	$(python) -m virtualenv --no-download ./env/
-	$(pip) install $$(for f in requirements*.txt; do echo "-r $$f"; done)
+	$(env_bin)/$(pip) install $$(for f in requirements*.txt; do echo "-r $$f"; done)
 	@touch env
 
 clean:

--- a/Makefile
+++ b/Makefile
@@ -14,7 +14,7 @@ echo:
 	@echo $($(var))
 
 env: requirements*.txt
-	$(python) -c "import virtualenv" || pip install virtualenv
+	$(python) -m pip --disable-pip-version-check install --user "virtualenv>=15.0.0"
 	$(python) -m virtualenv --no-download ./env/
 	$(pip) install $$(for f in requirements*.txt; do echo "-r $$f"; done)
 	@touch env

--- a/Makefile
+++ b/Makefile
@@ -15,8 +15,11 @@ py_test := $(with_tests_env) $(env_bin)/py.test
 echo:
 	@echo $($(var))
 
+_warning:
+	@echo -e "\nWarning: you're using an old version of python, you really should upgrade!\n"
+
 env: requirements*.txt
-	$(python) -m ensurepip $(install_where)
+	$(python) -m ensurepip $(install_where) || $(MAKE) --no-print-directory _warning
 	$(python) -m $(pip) install $(install_where) "virtualenv>=15.0.0"
 	$(python) -m virtualenv --no-download ./env/
 	$(env_bin)/$(pip) install $$(for f in requirements*.txt; do echo "-r $$f"; done)

--- a/README.md
+++ b/README.md
@@ -41,7 +41,6 @@ We interact with the database by writing raw SQL queries sent via the [postgres.
 Firstly, make sure you have the following dependencies installed:
 
 - python 2.7 ([we're working on porting to python 3](https://github.com/liberapay/liberapay.com/pull/88))
-- virtualenv
 - postgresql 9.4.5
 - make
 

--- a/README.md
+++ b/README.md
@@ -40,7 +40,7 @@ We interact with the database by writing raw SQL queries sent via the [postgres.
 
 Firstly, make sure you have the following dependencies installed:
 
-- python 2.7 ([we're working on porting to python 3](https://github.com/liberapay/liberapay.com/pull/88))
+- python ≥ 2.7.8 ([we're working on porting to python 3](https://github.com/liberapay/liberapay.com/pull/88))
 - postgresql 9.4.5
 - make
 

